### PR TITLE
TST: add CoW test for setitem with Series being set

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/issues/49473

We already have a test for setitem from https://github.com/pandas-dev/pandas/pull/50692/, but that only handled setting an array (actually, does that copy the array?), and not setting with a pandas object for which we could use lazy copy mechanism. 

This is only adding the test, we seem to actually do an eager copy, so room for optimization.

### CODE REVIEW

File: pandas/tests/copy_view/test_methods.py [hunk 1]
```diff
@@ -1313,6 +1313,33 @@ def test_isetitem(using_copy_on_write):
...
+    df.loc[0, "a"] = 0
+    tm.assert_series_equal(ser, ser_orig)
+
+    # mutating series doesn't update dataframe
+    df = DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    ser = Series([7, 8, 9])
+    df.isetitem(1, ser)  # This is inplace
+
+    ser.loc[0] = 0
+    expected = df = DataFrame({"a": [7, 8, 9], "b": np.array([4, 5, 6], dtype=dtype)})
```
Comment (phofl) [inline pandas/tests/copy_view/test_methods.py:1339]: I think you want to remove df here?

**Review by phofl — COMMENTED** [review]
small comment,

the # This is inplace comment is a bit confusing, the docstring says it will never try to write into the underlying array

**Comment by phofl** [comment]
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.isetitem.html

**Comment by jorisvandenbossche** [comment]
The comment in the test was a left-over from copying the previous setitem test, will remove.

But the docstring is actually confusing here, because it _is_ an inplace method in the sense that it is updating the _object_ inplace (we listed this method in the "always inplace" bucket in the draft PDEP). It's only not inplace in the sense that it will not update _values_ inplace.

**Comment by phofl** [comment]
Yeah good point, a clarification would probably help

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Fixed a small typo/copy paste issue. otherwise good to merge on greenish

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @jorisvandenbossche 

failures were unrelated

**Pull request merged**